sdk/digitaltwins/azure-digitaltwins-core/API design.md

```diff
@@ -319,19 +343,40 @@ These APIs are invoked via DigitalTwinsAsyncClient.
      * @return A REST response containing the application/json relationship corresponding to the provided relationshipId.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<String>> getRelationshipWithResponse(String digitalTwinId, String relationshipId) { }
+    public Mono<DigitalTwinsResponse<String>> getRelationshipWithResponse(String digitalTwinId, String relationshipId) { }
+
+    /**
+     * Gets a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to retrieve.
+     * @param modelClass The model class to convert the relationship to.
+     * @return The relationship corresponding to the provided relationshipId.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<T> getRelationship(String digitalTwinId, String relationshipId, Class<T> modelClass) { }
+
+    /**
+     * Gets a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to retrieve.
+     * @param modelClass The model class to convert the relationship to.
+     * @return A REST response containing the relationship corresponding to the provided relationshipId.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<DigitalTwinsResponse<T>> getRelationshipWithResponse(String digitalTwinId, String relationshipId, Class<T> modelClass) { }
 
     /**
      * Updates the properties of a relationship on a digital twin.
      *
      * @param digitalTwinId The Id of the source digital twin.
      * @param relationshipId The Id of the relationship to be updated.
      * @param relationshipUpdateOperations The application/json-patch+json operations to be performed on the specified digital twin's relationship.
-     * @param options The optional settings for this request.
      * @return An empty response.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
```

Comment:
do we need another Mono<void> updateRelationship that takes a RequestOption?

Reply:
No, we don't.

We have 2 overloads for `updateRelationship`:
1. `updateRelationship(String digitalTwinId, String relationshipId, List<Object> relationshipUpdateOperations)`
2. `updateRelationship(String digitalTwinId, String relationshipId, List<Object> relationshipUpdateOperations, RequestOptions options)`

The max overload one, with `RequestOptions` (2), will:
- async API -> return `withResponse`.
- sync API -> accept a `Context` input, and return `withResponse`.